sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareFileSeekableByteChannelWriteOptions.java

```diff
@@ -2,84 +2,43 @@
 // Licensed under the MIT License.
 package com.azure.storage.file.share.options;
 
-import com.azure.core.util.ExpandableStringEnum;
 import com.azure.core.util.logging.ClientLogger;
 import com.azure.storage.file.share.models.FileLastWrittenMode;
 import com.azure.storage.file.share.models.ShareRequestConditions;
-import com.fasterxml.jackson.annotation.JsonCreator;
-
-import java.util.Collection;
-import java.util.Objects;
 
 /**
  * Options for obtaining a {@link java.nio.channels.SeekableByteChannel} backed by an Azure Storage Share File.
  */
 public final class ShareFileSeekableByteChannelWriteOptions {
     private static final ClientLogger LOGGER = new ClientLogger(ShareFileSeekableByteChannelWriteOptions.class);
 
-    /**
-     * Mode to open the channel for writing.
-     */
-    public static final class WriteMode extends ExpandableStringEnum<WriteMode> {
-        /**
-         * Opens channel to an existing file for writing.
-         */
-        public static final WriteMode WRITE = fromString("Write");
-
-        /**
-         * Creates a new file for writing and opens the channel. If the file already exists, it will be overwritten.
-         * Requires a value be set with {@link ShareFileSeekableByteChannelWriteOptions#setFileSize(Long)}.
-         */
-        public static final WriteMode OVERWRITE = fromString("Overwrite");
-
-        /**
-         * Creates or finds a AccessTier from its string representation.
-         *
-         * @param name a name to look for.
-         * @return the corresponding AccessTier.
-         */
-        @JsonCreator
-        public static WriteMode fromString(String name) {
-            return fromString(name, WriteMode.class);
-        }
-
-        /**
-         * Gets known WriteMode values.
-         *
-         * @return known WriteMode values.
-         */
-        public static Collection<WriteMode> values() {
-            return values(WriteMode.class);
-        }
-    }
-
-    private final WriteMode channelMode;
+    private final boolean overwriteMode;
     private Long fileSize;
     private ShareRequestConditions requestConditions;
     private FileLastWrittenMode fileLastWrittenMode;
     private Long chunkSizeInBytes;
 
     /**
      * Options constructor.
-     * @param mode What usage mode to open the channel in.
+     * @param overwriteMode Whether to open the channel in write mode.
```

Comment:
```suggestion
     * @param overwriteMode If {@code true}, the channel will be opened in overwrite mode. Otherwise, the channel will be opened in write mode.
```